File: Packages/module-ai/Sources/ModuleAI/Application/UseCases/AIChat/LoadHistoryDetailUseCase.swift
//
//  LoadHistoryDetailUseCase.swift
//  ModuleAI
//
//  Created by Codex on 2026/1/8.
//

import Foundation

struct LoadHistoryDetailUseCase {
    private let repository: AIChatRepository

    init(repository: AIChatRepository) {
        self.repository = repository
    }

    func execute(historyId: Int) async throws -> [AIChat] {
        let record = try await repository.loadHistoryDetail(historyId)
        return AIChatHistoryMapper.makeChats(from: record.detailList ?? [])
    }
}

File: app-android/feature/ai-chat/src/main/java/com/aidatainsight/android/feature/aichat/application/usecase/LoadHistoryDetailUseCase.kt
package com.aidatainsight.android.feature.aichat.application.usecase

import com.aidatainsight.android.feature.aichat.application.AIChatApplicationMapper
import com.aidatainsight.android.feature.aichat.application.model.LoadHistoryDetailOutput
import com.aidatainsight.android.feature.aichat.domain.AIChatRepository

class LoadHistoryDetailUseCase(
    private val repository: AIChatRepository,
) {
    suspend operator fun invoke(historyId: Int): LoadHistoryDetailOutput {
        val record = repository.loadHistoryDetail(historyId)
        return LoadHistoryDetailOutput(
            messages = AIChatApplicationMapper.makeMessages(record.detailList.orEmpty()),
        )
    }
}
